model/testutil/stats.go

```diff
@@ -0,0 +1,257 @@
+package testutil
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/evergreen-ci/evergreen/apimodels"
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/evergreen/model/task"
+	"github.com/evergreen-ci/evergreen/model/testresult"
+	"github.com/stretchr/testify/suite"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+type TaskStatus struct {
+	Status         string
+	DetailsType    string
+	DetailsTimeout bool
+	TimeTaken      time.Duration
+}
+
+var Success100 = TaskStatus{"success", "test", false, 100 * 1000 * 1000 * 1000}
+var Success200 = TaskStatus{"success", "test", false, 200 * 1000 * 1000 * 1000}
+var TestFailed = TaskStatus{"failed", "test", false, 20}
+var Timeout = TaskStatus{"failed", "test", true, 100}
+var SetupFailed = TaskStatus{"failed", "setup", false, 10}
+var SystemFailed = TaskStatus{"failed", "system", false, 10}
+
+func ClearCollection(s suite.Suite, name string) {
+	err := db.Clear(name)
+	s.Require().NoError(err)
+}
+
+func InsertTask(s suite.Suite, project string, requester string, taskId string, execution int,
+	taskName string, variant string, distro string, createTime time.Time, status TaskStatus) {
+	details := apimodels.TaskEndDetail{
+		Status:   status.Status,
+		Type:     status.DetailsType,
+		TimedOut: status.DetailsTimeout,
+	}
+	newTask := task.Task{
+		Id:           taskId,
+		Execution:    execution,
+		Project:      project,
+		DisplayName:  taskName,
+		Requester:    requester,
+		BuildVariant: variant,
+		DistroId:     distro,
+		CreateTime:   createTime,
+		Status:       status.Status,
+		Details:      details,
+		TimeTaken:    status.TimeTaken,
+	}
+	err := newTask.Insert()
+	s.Require().NoError(err)
+}
+
+func InsertFinishedTask(s suite.Suite, project string, requester string, taskName string,
+	createTime time.Time, finishTime time.Time, execution int) task.Task {
+	newTask := task.Task{
+		Id:          bson.NewObjectId().Hex(),
```

Comment:
(my advice is to replace all these functions, but I'd also just let the driver create the Id field.

Reply:
I'm not using these function anymore, but am still creating the Id. It seems like when I omit the Id field, it is creating a task with an Id of "".